src/core/symbology-ng/qgssymbollayerutils.cpp

```diff
@@ -3986,8 +4003,26 @@ double QgsSymbolLayerUtils::rescaleUom( double size, QgsUnitTypes::RenderUnit un
           scale = 1 / 0.28;
           roundToUnit = true;
           break;
-        // we don't have a good case for map units, as pixel values won't change based on zoom
-        default:
+        case QgsUnitTypes::RenderInches:
+          scale = 1 / 0.28 * 25.4;
+          roundToUnit = true;
+          break;
+        case QgsUnitTypes::RenderPoints:
+          scale = 90. /* dots per inch according to OGC SLD */ / 72. /* points per inch */;
+          roundToUnit = true;
+          break;
+        case QgsUnitTypes::RenderPixels:
+          // pixel is pixel
+          scale = 1;
+          break;
+        case QgsUnitTypes::RenderMapUnits:
+        case QgsUnitTypes::RenderMetersInMapUnits:
+          // already handed via uom
+          scale = 1;
+          break;
+        case QgsUnitTypes::RenderPercentage:
+        case QgsUnitTypes::RenderUnknownUnit:
+          // these do not make sense and should not really reach here
```

Comment:
Actually it's better to just leave off the default - don't throw an exception. There'll be a build warning if new types are added and Travis will fail. That's safer then the exception route.